app/bundles/LeadBundle/EventListener/PageSubscriber.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * @copyright   2017 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\EventListener;
+
+use Mautic\CoreBundle\EventListener\CommonSubscriber;
+use Mautic\CoreBundle\Helper\BuilderTokenHelper;
+use Mautic\PageBundle\Event\PageBuilderEvent;
+use Mautic\PageBundle\Event\PageDisplayEvent;
+use Mautic\PageBundle\PageEvents;
+
+/**
+ * Class PageSubscriber.
+ */
+class PageSubscriber extends CommonSubscriber
+{
+    /**
+     * @var string
+     */
+    private static $contactFieldRegex = '{contactfield=(.*?)}';
+
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            PageEvents::PAGE_ON_BUILD   => ['onPageBuild', 0],
+            PageEvents::PAGE_ON_DISPLAY => ['onPageDisplay', -256],
+        ];
+    }
+
+    /**
+     * @param PageBuilderEvent $event
+     */
+    public function onPageBuild(PageBuilderEvent $event)
+    {
+        $tokenHelper = new BuilderTokenHelper($this->factory, 'lead.field', 'lead:fields', 'MauticLeadBundle');
+        // the permissions are for viewing contact data, not for managing contact fields
+        $tokenHelper->setPermissionSet(['lead:leads:viewown', 'lead:leads:viewother']);
+
+        if ($event->tokensRequested(self::$contactFieldRegex)) {
+            $event->addTokensFromHelper($tokenHelper, self::$contactFieldRegex, 'label', 'alias', true);
+        }
+    }
+
+    /**
+     * @param PageDisplayEvent $event
+     */
+    public function onPageDisplay(PageDisplayEvent $event)
+    {
+        $content = $event->getContent();
+        $regex   = '/\\'.str_replace('}', '\\}/', self::$contactFieldRegex);
+        preg_match_all($regex, $content, $matches);
+
+        /** @var \Mautic\LeadBundle\Model\LeadModel $leadModel */
+        $lead    = $this->factory->getModel('lead')->getCurrentLead();
```

Comment:
We're deprecating the use of Factory with 3.0. Please inject the lead model in the service config. This would add a constructor of:

```
public function __constructor(LeadModel $fieldModel)
{
    $this->leadModel = $leadModel;
}
```

and change this line to 

```
$lead = $this->leadModel->getCurrentLead();
```

and finally your service config

```
'mautic.lead.pagebundle.subscriber' => [
    'class' => Mautic\LeadBundle\EventListener\PageSubscriber::class,
    'arguments' => [
        'mautic.lead.model.lead'
    ],
],
```